File: practica/patrones-de-diseño/facade/impresionDocumentos/Documento.java
/**
 * Auto Generated Java Class.
 */
public class Documento {
  private int numeroDocumento;
  private String titulo;
  private boolean validado;
  private boolean impreso;
  private boolean firmado;
  
  public Documento(int numeroDocumento, String titulo){
    this.numeroDocumento = numeroDocumento;
    this.titulo = titulo;
    
    this.validado = false;
    this.impreso = false;
    this.firmado = false;
  }
  // faltan los setters y getters
  
  public String getTitulo(){
    return titulo;
  }
  
  public void setValidado(boolean nuevoEstado){
    this.validado = nuevoEstado;
  }
  
  public void setImpreso(boolean nuevoEstado){
    this.impreso = nuevoEstado;
  }
  
  public void setFirmado(boolean nuevoEstado){
    this.firmado = nuevoEstado;
  }
  
}

File: practica/patrones-de-diseño/typescript/facade/servicio-impresion/src/Documento.ts
export class Documento{
    private titulo:string
    private firmado:boolean
    private validado: boolean

    constructor(titulo:string){
        this.titulo = titulo
        this.firmado = false
        this.validado= false
    }

    public getTitulo():string{
        return this.titulo
    }

    public estaValidado(){
        return this.validado
    }

    public estaFirmado() {
        return this.validado
    }

    public setValidado(estado:boolean){
        this.validado=estado
    }

    public setFirmado(estado: boolean) {
        this.firmado= estado
    }
}
